bootstrap/handlers/auth_middleware.go

```diff
@@ -54,28 +60,32 @@ func VaultAuthenticationHandlerFunc(secretProvider interfaces.SecretProviderExt,
 				validToken, err := secretProvider.IsJWTValid(token)
 				if err != nil {
 					lc.Errorf("Error checking JWT validity: %v", err)
-					http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-					return
+					// set Response.Committed to true in order to rewrite the status code
+					w.Committed = false
+					return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
 				} else if !validToken {
 					lc.Warnf("Request to '%s' UNAUTHORIZED", r.URL.Path)
-					http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
-					return
+					// set Response.Committed to true in order to rewrite the status code
+					w.Committed = false
+					return echo.NewHTTPError(http.StatusUnauthorized, fmt.Sprintf("unauthorized request to '%s'", r.URL.Path))
 				}
 				lc.Debugf("Request to '%s' authorized", r.URL.Path)
-				inner(w, r)
-				return
+				return inner(c)
 			}
-			lc.Errorf("Unable to parse JWT for call to '%s'; unauthorized", r.URL.Path)
-			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+			err := fmt.Errorf("unable to parse JWT for call to '%s'; unauthorized", r.URL.Path)
+			lc.Errorf("%v", err)
+			// set Response.Committed to true in order to rewrite the status code
+			w.Committed = false
+			return echo.NewHTTPError(http.StatusUnauthorized, err.Error())
```

Comment:
For the access denied case, let's not return too much information to the caller as to why the access was denied.

Reply:
Thanks @bnevis-i, response message updated as you suggested.

Reply:
@lindseysimple  There was another one earlier that was also putting out err.Error().  I also added a comment there as well.